src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php

```diff
@@ -31,12 +33,14 @@ class SendFailedMessageForRetryListener implements EventSubscriberInterface
     private $sendersLocator;
     private $retryStrategyLocator;
     private $logger;
+    private $eventDispatcher;
 
-    public function __construct(ContainerInterface $sendersLocator, ContainerInterface $retryStrategyLocator, LoggerInterface $logger = null)
+    public function __construct(ContainerInterface $sendersLocator, ContainerInterface $retryStrategyLocator, LoggerInterface $logger = null, EventDispatcherInterface $eventDispatcher = null)
     {
         $this->sendersLocator = $sendersLocator;
         $this->retryStrategyLocator = $retryStrategyLocator;
         $this->logger = $logger;
+        $this->eventDispatcher = $eventDispatcher;
```

Comment:
We're missing the service wiring to actually inject the new argument

Reply:
indeed